src/data/boards.js

```diff
@@ -0,0 +1,190 @@
+const mongoCollections = require('../config/mongoCollections');
+const boards = mongoCollections.boards;
+const users = mongoCollections.users;
+const {colors} = require('../public/constants/index');
+const {ObjectId} = require('mongodb');
+const error_handler = require('../errors/error-handler'); 
+
+function newFields(newBoard, oldBoard) {
+  fields = {};
+  if(newBoard.boardName && newBoard.boardName !== oldBoard.boardName) {
+    if(!error_handler.checkNonEmptyString(newBoard.boardName))
+      throw new Error("Board name must not be empty.");
+    fields.boardName = newBoard.boardName;
+  }
+  if(newBoard.boardColor && newBoard.boardColor !== oldBoard.boardColor) {
+    if(!error_handler.checkColor(newBoard.boardColor))
+      throw new Error("Board color must be valid hex code.");
+    fields.boardColor = newBoard.boardColor;
+  }
+  if(newBoard.description && newBoard.description !== oldBoard.description) {
+    if(!error_handler.checkNonEmptyString(newBoard.description))
+      throw new Error("Board description must not be empty.");
+    fields.description = newBoard.description;
+  }
+
+  return fields;
+}
+
+
+module.exports = {
+  /**
+   * Get a board with a specific ID.
+   * @param {String} id The board's id.
+   * @returns A board.
+   */
+  readByID: async (id) => {
```

Comment:
style: I think general style rules follow that when camelcasing with 'ID' it should be 'Id' so this function would be called: readById(). Not going to prevent merging for this change, but something to consider.